clients/fossid-webapp/src/main/kotlin/Extensions.kt

```diff
@@ -302,3 +309,54 @@ suspend fun FossIdRestService.createIgnoreRule(
             "apply_to" to scope.name.lowercase()
         )
     )
+
+suspend fun FossIdRestService.generateReport(
+    user: String,
+    apiKey: String,
+    scanCode: String,
+    reportType: ReportType,
+    selectionType: SelectionType
+): File {
+    val response = generateReport(
+        PostRequestBody(
+            "generate_report",
+            SCAN_GROUP,
+            user,
+            apiKey,
+            "scan_code" to scanCode,
+            "report_type" to reportType.name.lowercase(),
+            "selection_type" to selectionType.name.lowercase()
+        )
+    )
+
+    require(response.isSuccessful) {
+        """ ${response.code()} - ${response.message()}
+            ${response.errorBody()?.string()}
+        """.trimIndent()
+    }
+
+    // The API is quirky here: if the report type is HTML, the result is returned as a standard HTML response, without
+    // filename. For any other report types, the result is returned as an attachment with a filename in the headers
+    // (even for report types generating HTML such as DYNAMIC).
+    val fileName = if (reportType == ReportType.HTML) "${UUID.randomUUID()}-report.html"
+    else {
+        response.headers()["Content-disposition"]?.split(';')?.firstNotNullOfOrNull {
+            it.trim().withoutPrefix("filename=")?.removeSurrounding("\"")
+        } ?: throw IllegalStateException("Cannot determine name of the report")
+    }
+
+    val tempDir = createTempDirectory("fossid_report").toFile()
+        .apply { deleteOnExit() }
+
+    return response.body()?.source().use {
+        File(tempDir, fileName).apply {
+            it?.inputStream()?.copyTo(outputStream())
+        }
+    }
+}
+
+/**
+ * If this string starts with [prefix], return the string without the prefix, otherwise return [missingPrefixValue].
+ */
+fun String?.withoutPrefix(prefix: String, missingPrefixValue: () -> String? = { null }): String? =
+    this?.removePrefix(prefix)?.takeIf { it != this } ?: missingPrefixValue()
```

Comment:
I wonder whether we should start to allow a dependency on ORT's `common-utils` in `clients` to avoid such code duplication...

Reply:
I notice you removed the dependency in a previous commit, hence the duplication....

Reply:
Do you mean 5423682cb5a9221b1b5d0b1c06cb214c81dfdf05? That code shouldn't even have compiled, as there was (and is) no dependency on ORT's libraries in `fossid-webapp`'s `build.gradle.kts`. And in the past [we wanted it to be like that](https://github.com/oss-review-toolkit/ort/blob/main/clients/README.md). However, since the split of ORT's utility modules into `common` and `core`, I'm less against adding a dependency from the clients on ORT `common-utils`. What do you think @mnonnenmacher?

Reply:
I would be fine with adding that dependency. However, it could also be done after this PR to not block it.

Reply:
Yes, agreed, we should do that afterwards and remove duplicate code from *all* client libraries at once in favor of introducing that dependency.